### PR TITLE
feat(datepicker): add date range select

### PR DESCRIPTION
**Pull request by nadiia**

### CODE REVIEW

**Comment by chasestarr** [comment]
not sure if you're still working on this pr, but it looks like the non-range date picker throws an error. also seeing that keyboard nav still works, but i cannot use enter/space to select a date

File: src/datepicker/styled-components.js [hunk 1]
```diff
@@ -110,9 +131,52 @@ export const StyledDay = styled('div', (props: SharedStylePropsT) => {
         : 'inherit',
     backgroundColor: $selected
       ? colors.primary
-      : $isHovered || $isHighlighted
+      : $isHovered || $isHighlighted || $pseudoHighlighted || $pseudoSelected
         ? colors.primary100
         : 'transparent',
-    borderRadius: borders.useRoundedCorners ? borders.radius200 : '0px',
+    ...(borders.useRoundedCorners
```
Comment (chasestarr) [inline src/datepicker/styled-components.js]: could this be broken out into its own function and unit tested? seems to work fine in practice though

Comment (chasestarr) [reply src/datepicker/styled-components.js]: what i'm concerned about is causing regressions on tangentially related code changes

File: src/datepicker/examples.js [hunk 2]
```diff
@@ -62,14 +74,15 @@ class DatepickerWithInput extends React.Component<
...
+            isRange={this.props.isRange}
             onSelect={this.onChange}
           />
         }
       >
         <Input
           value={this.state.formattedValue}
           onFocus={this.open}
-          onBlur={this.close}
+          // onBlur={this.close}
```
Comment (chasestarr) [inline src/datepicker/examples.js]:
```suggestion        
```

File: src/datepicker/examples.js [hunk 3]
```diff
@@ -14,6 +14,7 @@ import {
 import {Popover, PLACEMENT} from '../popover/index.js';
 import {Input} from '../input/index.js';
 import tests from './examples-list.js';
+import {highlight} from 'lowlight';
```
Comment (chasestarr) [inline src/datepicker/examples.js]: is this used?

```suggestion
```

File: src/datepicker/day.js [hunk 4]
```diff
@@ -101,15 +125,77 @@ export default class Day extends React.Component<DayPropsT, DayStateT> {
...
+    }
+  }
+
   getSharedProps() {
+    const {date} = this.props;
+    const $isHighlighted = isSameDay(date, this.props.highlightedDate);
+    const $selected = this.isSelected(date);
+    const $hasRangeHighlighted =
+      this.props.isRange &&
+      this.props.value.length === 1 &&
```
Comment (chasestarr) [inline src/datepicker/day.js]: examples on netlify are breaking here or on L185
https://deploy-preview-708--baseui.netlify.com/?selectedKind=Unstable_Datepicker&selectedStory=Stateless%20datepicker&full=0&addons=1&stories=1&panelRight=0&addonPanel=REACT_STORYBOOK%2Freadme%2Fpanel

File: src/datepicker/examples.js (hunk 3 above)
Comment (nadiia) [reply src/datepicker/examples.js]: That's my editor being helpful by automatically adding imports

File: src/datepicker/day.js (hunk 4 above)
Comment (nadiia) [reply src/datepicker/day.js]: fixed this one

**Comment by nadiia** [comment]
> Code changes lgtm, but keyboard nav seems to be broken when selecting a date.

Thanks for catching it, was a result of a prop removal - fixed!

**Pull request merged**